staging/src/k8s.io/kubectl/pkg/cmd/create/create_secret_tls.go

```diff
@@ -228,16 +233,101 @@ func (o *CreateSecretTLSOptions) createSecretTLS() (*corev1.Secret, error) {
 	if err != nil {
 		return nil, err
 	}
+	var tlsCertAuthority []byte
+	if len(o.CertAuthority) != 0 {
+		tlsCertAuthority, err = readFile(o.CertAuthority)
+		if err != nil {
+			return nil, err
+		}
+	}
 	if _, err := tls.X509KeyPair(tlsCert, tlsKey); err != nil {
 		return nil, err
 	}
+
+	// validate that cert is a valid certificate
+	rest := tlsCert
+	var _tlsCert *x509.Certificate
+	var block *pem.Block
+	for {
+		block, rest = pem.Decode(rest)
+		if block == nil || block.Type != "CERTIFICATE" {
+			return nil, fmt.Errorf("failed to decode PEM block containing certificate.")
+		}
+		_tlsCert, err = x509.ParseCertificate(block.Bytes)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse PEM block in certificate: %s", err)
+		}
+		if len(rest) == 0 {
+			break
+		}
+	}
+
+	// validate certAuthority is a valid certificate
+	rest = tlsKey
+	for {
+		block, rest = pem.Decode(rest)
+		if block == nil {
+			return nil, fmt.Errorf("failed to decode PEM block containing private key.")
+		}
+		if block.Type == "RSA PRIVATE KEY" {
+			_, err := x509.ParsePKCS1PrivateKey(block.Bytes)
+			if err != nil {
+				return nil, fmt.Errorf("failed to parse PEM block in RSA private key, %s", err)
+			}
+		} else if block.Type == "EC PRIVATE KEY" {
+			_, err := x509.ParseECPrivateKey(block.Bytes)
+			if err != nil {
+				return nil, fmt.Errorf("failed to parse PEM block in EC private key: %s", err)
+			}
+		} else {
+			return nil, fmt.Errorf("unknown PEM block in private key: %s.", block.Type)
+		}
+		if len(rest) == 0 {
+			break
+		}
+	}
```

Comment:
hey, thanks for adding this but I guess this is already verified by 

```
	if _, err := tls.X509KeyPair(tlsCert, tlsKey); err != nil {
		return nil, err
	}
```

or not? :)

I guess so, if you take a look into tls.X509KeyPair it does the decode, etc etc and fails if the cert is not a pair of the key, so I guess we are fine without this :)

Reply:
Even better!  Will be gone in the next push.